regrid/regrid.py

```diff
@@ -0,0 +1,251 @@
+
+        # interpolation weights files
+        wgtsfile_t_to_t = (
+            f"{self.output_path}/weights-{self.ires}.Ct.{self.ores}.Ct.bilinear.nc"
+        )
+        wgtsfile_u_to_t = (
+            f"{self.output_path}/weights-{self.ires}.Cu.{self.ires}.Ct.bilinear.nc"
+        )
+        wgtsfile_v_to_t = (
+            f"{self.output_path}/weights-{self.ires}.Cv.{self.ires}.Ct.bilinear.nc"
+        )
+
+        # define regridding instances
+        reuse = os.path.exists(wgtsfile_t_to_t)
+        self.rg_tt = xe.Regridder(
+            ds_in_t,
+            grid_out,
+            self.interp_method,
+            periodic=True,
+            reuse_weights=reuse,
+            filename=wgtsfile_t_to_t,
+        )
+        reuse = os.path.exists(wgtsfile_u_to_t)
+        self.rg_ut = xe.Regridder(
+            ds_in_u,
+            ds_in_t,
+            self.interp_method,
+            periodic=True,
+            reuse_weights=reuse,
+            filename=wgtsfile_u_to_t,
+        )
+        reuse = os.path.exists(wgtsfile_v_to_t)
+        self.rg_vt = xe.Regridder(
+            ds_in_v,
+            ds_in_t,
+            self.interp_method,
+            periodic=True,
+            reuse_weights=reuse,
+            filename=wgtsfile_v_to_t,
+        )
+
+    def regrid(self, file):
+        """Regrid a single ocean file"""
+        print(f"Regridding file: {file}")
+        dtg = file[-16:-3]
+        ds_in = xr.open_dataset(file)
+        ds_out = []
+
+        for var in list(ds_in.keys()):
+            if len(ds_in[var].coords) > 2:
+                coords = ds_in[var].coords.to_index()
+
+                # choose regridding type
+                if coords.names[0] != "time":
+                    raise ValueError("First coordinate should be time")
+                else:
+                    variable_map = {
+                        "SSU": ("SSV", "U"),
+                        "SSV": (None, "skip"),
+                        "uo": ("vo", "U"),
+                        "vo": (None, "skip"),
+                        "taux": ("tauy", "U"),
+                        "tauy": (None, "skip"),
+                    }
+                    if var in variable_map.keys():
+                        var2, pos = variable_map[var]
+                    else:
+                        var2, pos = (None, "T")
+
+                # 3-dimensional data
+                is_3d = False
+                if coords.names[1] == "z_l" or coords.names[1] == "zl":
+                    dims = ["time", "lev", "lat", "lon"]
+                    is_3d = True
+                else:
+                    dims = ["time", "lat", "lon"]
+
+                if pos == "T":
+                    interp_out = self.rg_tt(ds_in[var].values)
+                    da_out = xr.DataArray(interp_out, dims=dims)
+                    da_out.attrs["long_name"] = ds_in[var].long_name
+                    da_out.attrs["units"] = ds_in[var].units

```

Comment:
Sorry for all the nitpicks ... 
```suggestion
                    da_out = xr.DataArray(
                        interp_out,
                        dims=dims,
                        attrs={
                            "long_name": ds_in[var].long_name,
                            "units": ds_in[var].units,
                    )
```